media.go

```diff
@@ -313,6 +340,15 @@ func (pc *producerConsumer[T, U]) stop() {
 }
 
 func (pc *producerConsumer[T, U]) stopOne() {
+	// We want to create a new span that lasts the scope of this function and no
+	// further, so we have to modify the cancelCtx and then revert it at the end
+	// of the function to what it was before.
+	originalCancelCtx := pc.cancelCtx
```

Comment:
Nice! I didn't think to try using a temp variable to change and revert the context like this. Tried it out [here](https://go.dev/play/p/VQgmNw0mvGZ), and it works!

I'm curious about what happens if you remove the `originalCancelCtx`. I'd guess that the span would start on line 348 and then stop at the end of the function. That seems similar to what's going on now. Is that not the case or is there another error you ran into?

Reply:
Now that I look at it again you're definitely right, I wouldn't have to add the temp variable there